File: registratorapp/src/main/java/de/weimarnetz/registrator/services/MacAddressService.java
package de.weimarnetz.registrator.services;

import org.springframework.stereotype.Component;

import de.weimarnetz.registrator.exceptions.InvalidMacAddressException;

@Component
public class MacAddressService {

    public boolean isValidMacAddress(String macAddress) {
        return macAddress.matches("^([a-fA-F0-9]{2})([a-f0-9A-F]{2})([a-f0-9A-F]{2})([a-f0-9A-F]{2})([a-f0-9A-F]{2})([a-f0-9A-F]{2})$|^([a-f0-9A-F]{2}):([a-f0-9A-F]{2}):([a-f0-9A-F]{2}):([a-f0-9A-F]{2}):([a-f0-9A-F]{2}):([a-f0-9A-F]{2})$|^([a-f0-9A-F]{2})-([a-f0-9A-F]{2})-([a-f0-9A-F]{2})-([a-f0-9A-F]{2})-([a-f0-9A-F]{2})-([a-f0-9A-F]{2})$|^([0-9A-Fa-f]{4})\\.([0-9A-Fa-f]{4})\\.([0-9A-Fa-f]{4})$");
    }

    public String normalizeMacAddress(String macAddress) throws InvalidMacAddressException {
        if (!isValidMacAddress(macAddress)) {
            throw new InvalidMacAddressException("Mac Address invalid: " + macAddress);
        }
        return macAddress.replace(":", "").replace("-", "").toLowerCase();
    }
}

File: src/main/java/de/weimarnetz/registrator/services/MacAddressService.kt
package de.weimarnetz.registrator.services

import de.weimarnetz.registrator.exceptions.InvalidMacAddressException
import org.springframework.stereotype.Component
import java.util.Locale

@Component
class MacAddressService {
    fun isValidMacAddress(macAddress: String?): Boolean {
        return macAddress?.matches("^([a-fA-F\\d]{2})([a-f\\dA-F]{2})([a-f\\dA-F]{2})([a-f\\dA-F]{2})([a-f\\dA-F]{2})([a-f\\dA-F]{2})$|^([a-f\\dA-F]{2}):([a-f\\dA-F]{2}):([a-f\\dA-F]{2}):([a-f\\dA-F]{2}):([a-f\\dA-F]{2}):([a-f\\dA-F]{2})$|^([a-f\\dA-F]{2})-([a-f\\dA-F]{2})-([a-f\\dA-F]{2})-([a-f\\dA-F]{2})-([a-f\\dA-F]{2})-([a-f\\dA-F]{2})$|^([\\dA-Fa-f]{4})\\.([\\dA-Fa-f]{4})\\.([\\dA-Fa-f]{4})$".toRegex())
            ?: false
    }

    @Throws(InvalidMacAddressException::class)
    fun normalizeMacAddress(macAddress: String?): String {
        if (!isValidMacAddress(macAddress)) {
            throw InvalidMacAddressException()
        }
        return macAddress!!.replace(":", "").replace("-", "").lowercase(Locale.getDefault())
    }
}